docs/how-to/hip_runtime_api/multi_device.rst

```diff
@@ -0,0 +1,287 @@
+            std::cout << "  Number of Multiprocessors: " << deviceProp.multiProcessorCount << std::endl;
+            std::cout << "  Max Threads Dimensions: [" 
+                    << deviceProp.maxThreadsDim[0] << ", "
+                    << deviceProp.maxThreadsDim[1] << ", "
+                    << deviceProp.maxThreadsDim[2] << "]" << std::endl;
+            std::cout << "  Max Grid Size: [" 
+                    << deviceProp.maxGridSize[0] << ", "
+                    << deviceProp.maxGridSize[1] << ", "
+                    << deviceProp.maxGridSize[2] << "]" << std::endl;
+            std::cout << std::endl;
+        }
+
+        return 0;
+    }
+
+.. _multi-device_selection:
+
+Device selection
+===============================================================================
+
+Once you have enumerated the available GPUs, the next step is to select a
+specific device for computation. This involves setting the active GPU that will
+execute subsequent operations. This step is crucial in multi-GPU systems where
+different GPUs might have different capabilities or workloads. By selecting the
+appropriate device, you ensure that the computational tasks are directed to the
+correct GPU, optimizing performance and resource utilization.
+
+.. code-block:: cpp
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    #define HIP_CHECK(expression)                \
+    {                                            \
+        const hipError_t status = expression;    \
+        if (status != hipSuccess) {              \
+            std::cerr << "HIP error " << status  \
+                    << ": " << hipGetErrorString(status) \
+                    << " at " << __FILE__ << ":" \
+                    << __LINE__ << std::endl;  \
+            exit(status);                        \
+        }                                        \
+    }
+
+    __global__ void simpleKernel(double *data)
+    {
+        int idx = blockIdx.x * blockDim.x + threadIdx.x;
+        data[idx] = idx * 2.0;
+    }
+
+    int main()
+    {
+        double* deviceData0;
+        double* deviceData1;
+        size_t  size = 1024 * sizeof(*deviceData0);
+
+        int deviceId0 = 0;
+        int deviceId1 = 1;
+
+        // Set device 0 and perform operations
+        HIP_CHECK(hipSetDevice(deviceId0)); // Set device 0 as current
+        HIP_CHECK(hipMalloc(&deviceData0, size)); // Allocate memory on device 0
+        simpleKernel<<<1000, 128>>>(deviceData0); // Launch kernel on device 0
+        HIP_CHECK(hipDeviceSynchronize());
+
+        // Set device 1 and perform operations
+        HIP_CHECK(hipSetDevice(deviceId1)); // Set device 1 as current
+        HIP_CHECK(hipMalloc(&deviceData1, size)); // Allocate memory on device 1
+        simpleKernel<<<1000, 128>>>(deviceData1); // Launch kernel on device 1
+        HIP_CHECK(hipDeviceSynchronize());
+
+        // Attempt to use deviceData0 on device 1 (This will not work as deviceData0 is allocated on device 0)
```

Comment:
Is there some shared memory or global memory that would allow device 1 to access data from device 0? Or perhaps some mechanism to copy data from device 0 to device 1? Is this worth mentioning here?

Reply:
Maybe mention the Peer to Peer section below?